### PR TITLE
Switch to HTTPX to address regression in Python 3.10

### PR DESCRIPTION
**Pull request by 3asm**
As reported in the following issue https://github.com/python/cpython/issues/110467, we are encountering a regression in python 3.10 and newer versions.

The issue do not appear on `httpx` , hence the change.

### CODE REVIEW

File: src/ostorlab/cli/agent/healthcheck/healthcheck.py [hunk 1]
```diff
@@ -37,12 +37,12 @@ def healthcheck(host: str, port: int, https: bool = False) -> None:
...
     try:
-        response = requests.get(url, timeout=10)
+        response = httpx.get(url, timeout=10)
         if response.status_code != 200:
             console.error(f"Response status code is {response.status_code}")
             raise click.exceptions.Exit(2)
         else:
             console.success(f"Healthcheck OK on {url}!")
-    except requests.ConnectionError as e:
+    except httpx.HTTPError as e:
```
Comment (oussamaessaji) [inline src/ostorlab/cli/agent/healthcheck/healthcheck.py:46]:
```suggestion
    except httpx.ConnectError as e:
```

**Review by oussamaessaji — CHANGES_REQUESTED** [review]

File: tests/cli/docker_requirements_checker_test.py [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-from requests_mock import mocker as req_mocker
 
 from ostorlab import exceptions
 from ostorlab.cli import docker_requirements_checker
 
 
 @pytest.mark.docker
 def testRuntime_WhenCantInitSwarm_shouldRetry(
-    mocker: plugin.MockerFixture, requests_mock: req_mocker.Mocker
+    mocker: plugin.MockerFixture, httpx_mock
```
Comment (oussamaessaji) [inline tests/cli/docker_requirements_checker_test.py:11]: type please

File: tests/agent/mixins/agent_mq_mixin_test.py [hunk 3]
```diff
@@ -130,7 +129,7 @@ async def testClient_whenClientDisconnects_messageIsNotLost(mocker, mq_service):
 
 
 def testMqSendMessage_onConnectionResetError_shouldRetriesAndReraise(
-    mocker: req_mocker.Mocker,
+    mocker,
```
Comment (elyousfi5) [inline tests/agent/mixins/agent_mq_mixin_test.py:132]: why remove the typing ?

File: tests/apis/create_asset_test.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 """Unittests for the create asset API request."""
+
```
Comment (deadly-panda) [inline tests/apis/create_asset_test.py:2]:
```suggestion
"""Unit tests for the create asset API request."""

```

File: tests/runtimes/definitions_test.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
 """Unittest for runtime definitions."""
+
```
Comment (deadly-panda) [inline tests/runtimes/definitions_test.py:2]:
```suggestion
"""Unit test for runtime definitions."""

```

File: tests/runtimes/lite_local/runtime_test.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 """Unittest for lite local runtime."""
+
```
Comment (deadly-panda) [inline tests/runtimes/lite_local/runtime_test.py:2]:
```suggestion
"""Unit test for lite local runtime."""

```

File: tests/runtimes/local/agent_runtime_test.py [hunk 7]
```diff
@@ -1,4 +1,5 @@
 """Unittest for agent runtime."""
+
```
Comment (deadly-panda) [inline tests/runtimes/local/agent_runtime_test.py:2]: Apply to the rest please.

**Review by amine3 — DISMISSED** [review]

File: src/ostorlab/apis/runners/runner.py [hunk 8]
```diff
@@ -60,19 +60,13 @@ def execute(self, request: api_request.APIRequest) -> Dict[str, Any]:
...
-            timeout=REQUEST_TIMEOUT,
-        )
+        with httpx.Client(proxy=self._proxy, verify=self._verify) as client:
+            return client.post(
+                self.endpoint,
+                data=request.data,
+                files=request.files,
+                headers=headers,
+                timeout=REQUEST_TIMEOUT,
+            )
```
Comment (deadly-panda) [inline src/ostorlab/apis/runners/runner.py:72]: This method in the baseRunner seem odd, the public/authenticated runners both implement theirs, only the loginRunner uses this one.

File: tests/cli/agent/install/install_test.py [hunk 9]
```diff
@@ -40,9 +37,15 @@ def testAgentInstallCLI_whenAgentDoesNotExist_commandExitsWithError(
...
-    )
+
+    def _custom_matcher(request):
+        matcher = re.compile(r"http\+docker://(.*)/version")
+        if re.match(matcher, str(request.url)) is not None:
+            return httpx.Response(200, json={"ApiVersion": "1.42"})
+        if str(request.url) == public_runner.PUBLIC_GRAPHQL_ENDPOINT:
+            return httpx.Response(200, json=api_call_response)
+
+    httpx_mock.add_callback(_custom_matcher)
```
Comment (deadly-panda) [inline tests/cli/agent/install/install_test.py:48]: Jamil!

File: tests/cli/auth/test_login.py [hunk 10]
```diff
@@ -81,16 +88,25 @@ def testOstorlabAuthLoginCLI_whenValidLoginCredentialsAreProvidedWithoutOtp_prom
...
+            "status_code": 400,
+        },
+        {"json": token_dict, "status_code": 200},
+    ]
+    response_idx = 0
+
+    def _request_callback(request, response_idx):
+        response = httpx.Response(**responses[response_idx])
+        response_idx += 1
+        return response
```
Comment (deadly-panda) [inline tests/cli/auth/test_login.py:104]: Can't u achieve the same this by just adding multiple responses? same in `tests/runtimes/cloud/runtime_test.py:83` ?

File: tests/scanner/scan_handler_test.py [hunk 11]
```diff
@@ -1,17 +1,17 @@
...
 from ostorlab.scanner import scan_handler
 from ostorlab.scanner import scanner_conf
-from ostorlab.apis.runners import authenticated_runner
 from ostorlab.utils import scanner_state_reporter
 
 
 @pytest.mark.asyncio
 async def testConnectNats_whenScannerConfig_subscribeNatsWithStartAgentScan(
-    requests_mock, mocker, data_start_agent_scan
+    mocker, data_start_agent_scan
```
Comment (deadly-panda) [inline tests/scanner/scan_handler_test.py:14]: this was not needed?

**Review by deadly-panda — CHANGES_REQUESTED** [review]

File: tests/cli/auth/test_login.py (hunk 10 above)
Comment (3asm) [reply tests/cli/auth/test_login.py:104]: It doesn't work.

**Review by amine3 — APPROVED** [review]

**Pull request merged**